MAGECLOUD-3811 Add section about generic webhooks (#4963)
magento/devdocs: guides/v2.2/cloud/integrations/cloud-integrations.md

+----------+--------------+---------------------------------------------------------------------------+
```
{: .no-copy}

## Generic webhooks

You can capture events and `POST` JSON messages that report activity to a _webhook_ URL. Add a webhook URL using the following syntax:

```bash
magento-cloud integration:add --type=webhook --url=https://hook-url.example.com
```

-  `type`—Specify the `webhook` integration type.
-  `url`—Provide the webhook URL that can receive JSON messages.

> The sample response shows a series of prompts that provide an opportunity to customize the integration. Using the default (blank) response sends all events on all environments in a project.

Reviewer: ```suggestion
The sample response shows a series of prompts that provide an opportunity to customize the integration. Using the default (blank) response sends messages about all events on all environments in a project.
```

Reviewer: or "posts"

> You can customize the integration to report specific [events](#events-to-report), such as pushing code to a branch. For example, you can specify the `environment.push` event sends a message when a user pushes code to a branch:

Reviewer: ```suggestion
You can customize the integration to report specific [events](#events-to-report), such as pushing code to a branch. For example, you can specify the `environment.push` event to send a message when a user pushes code to a branch:
```

```terminal
Events to report (--events)

> A list of events to report, e.g. environment.push

Reviewer: ```suggestion
A list of events to report, for example `environment.push`
```

Reviewer: oh -- I guess the `e.g.` is in the code :-(

Default: *
Enter comma-separated values (or leave this blank)
>
```
{: .no-copy}

> You can filter states, which can be `pending`, `in_progress`, or `complete`:

Reviewer: This is OK, but it seems a little unclear. Consider:

 filter event reporting by states, which can include ....  and `complete" (Since you can enter more than one value )

```
{: .no-copy}

And you can _include_ or _exclude_ messages for specific environments:

```terminal
Included environments (--environments)
The environment IDs to include
Default: *
Enter comma-separated values (or leave this blank)
>

Excluded environments (--excluded-environments)
The environment IDs to exclude
Enter comma-separated values (or leave this blank)
>
```
{: .no-copy}

When the integration is complete, you receive a summary of the values:

```terminal
Created integration integration-ID (type: webhook)
+-----------------------+------------------------------+
| Property              | Value                        |
+-----------------------+------------------------------+
| id                    | integration-ID               |
| type                  | webhook                      |
| events                | - '*'                        |
| environments          | - '*'                        |
| excluded_environments | {  }                         |
| states                | - complete                   |
| url                   | https://hook-url.example.com |
+-----------------------+------------------------------+
```
{: .no-copy}

### Update existing integration

> You can update an existing integration. For example, to change the states from `complete` to `pending` using the following:

Reviewer: ```suggestion
You can update an existing integration. For example, change the states from `complete` to `pending` using the following:
```

| environments          | - '*'                        |
| excluded_environments | {  }                         |
| states                | - pending                    |
| url                   | https://hook-url.example.com |
+-----------------------+------------------------------+
```
{: .no-copy}

### Events to report

Event | Description
----- | :-----------
`environment.access.add`    | A new user has been granted access to the environment
`environment.access.remove` | A user has been removed from the environment
`environment.activate`      | A branch has been "activated" with an environment
`environment.backup`        | A user triggered a snapshot
`environment.branch`        | A new branch has been created using the management console
`environment.deactivate`    | A branch has been "deactivated". The code is still there but the environment was destroyed
`environment.delete`        | A branch has been deleted
`environment.initialize`    | The master branch of the project initialized with a first commit
`environment.merge`         | An active branch has been merged using the management console or API
`environment.push`          | A user pushed code to a branch

> `environment.restore`       | A user restored a snapshot

Reviewer: Inconsistent punctuation on descriptions in table.

`environment.route.create`  | A new route has been created using the management console.
`environment.route.delete`  | A route has been deleted using the management console.
`environment.route.update`  | A route has been modified using the management console.
`environment.subscription.update` | The master environment has been resized because the subscription has changed. There are no content changes.
`environment.synchronize`   | An environment has had data or code re-copied from its parent environment.
`environment.update.http_access` | HTTP access rules for an environment have been modified.
`environment.update.restrict_robots` | The block-all-robots feature has been enabled or disabled.
`environment.update.smtp`   | Sending of emails has been enabled or disabled for an environment.

> `environment.variable.create` | A new variable has been created.

Reviewer: ```suggestion
`environment.variable.create` | A variable has been created.
```

`environment.variable.delete` | A variable has been deleted.
`environment.variable.update` | A variable has been modified.

> `project.domain.create`     | A new domain has been created and added to the project.

Reviewer: ```suggestion
`project.domain.create`     | A domain has been created and added to the project.
```
